typescript.js

```diff
@@ -0,0 +1,68 @@
+module.exports = {
+    extends: './index',
+    parser: '@typescript-eslint/parser',
+    rules: {
```

Comment:
I don't quite follow where this initial set of rules came from. At first I thought it was our JS rules in `index.js` but I don't think that's the case because I see lots of differences. Can someone help me understand?

Reply:
They are from [here](https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin#extension-rules). Their configurations should be equivalent to our ```index.js``` or the airbnb base. Can you point out an example of an incorrect one? Maybe I messed up or we're comparing differently. For example, brace-style is from [here](https://github.com/airbnb/javascript/blob/63098cbb6c05376dbefc9a91351f5727540c1ce1/packages/eslint-config-airbnb-base/rules/style.js#L21) while comma-dangle is from [here](https://github.com/ni/javascript-styleguide/blob/4e4d8426f1a6269f29c99682aade36538f1cedd1/index.js#L21).

Reply:
Ah, I understand better now. It was just confusion on my part. I didn't notice that this extends `index.js`. So I thought there would be a 1-1 mapping from rules in our `index.js` to rules in this file and I got confused when scanning and saw that some rules from `index.js` didn't appear here at all (which I now understand is because the config in `index` can apply to TypeScript code unchanged) and some rules appeared here multiple times (which I now understand is to turn off the original and to replace it with one from @typescript-eslint).

After wrapping my head around it, I didn't find any examples configured incorrectly (but I did more of a spot check than a comprehensive audit).